elektronLogging/clsfnTaskDistOverheadLogger.go

```diff
@@ -0,0 +1,50 @@
+package elektronLogging
+
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"strings"
+)
+
+type ClsfnTaskDistOverheadLogger struct {
+	LoggerImpl
+}
+
+func NewClsfnTaskDistOverheadLogger(logType int, prefix string) *ClsfnTaskDistOverheadLogger {
+	cLog := &ClsfnTaskDistOverheadLogger{}
+	cLog.Type = logType
+	cLog.SetLogFile(prefix)
+	return cLog
+}
+
+func (cLog *ClsfnTaskDistOverheadLogger) Log(logType int, level log.Level, logData log.Fields, message string) {
+	if cLog.Type == logType {
+
+		logger.SetLevel(level)
+
+		if cLog.AllowOnConsole {
+			logger.SetOutput(os.Stdout)
+			logger.WithFields(logData).Println(message)
+		}
+
+		logger.SetOutput(cLog.LogFileName)
+		logger.WithFields(logData).Println(message)
+	}
+	if cLog.next != nil {
+		cLog.next.Log(logType, level, logData, message)
+	}
+}
+
+func (cLog *ClsfnTaskDistOverheadLogger) SetLogFile(prefix string) {
+
+	tskDistLogPrefix := strings.Join([]string{prefix, config.TaskDistConfig.FilenameExtension}, "")
+	if logDir != "" {
+		tskDistLogPrefix = strings.Join([]string{logDir, tskDistLogPrefix}, "/")
```

Comment:
You should use [filepath.Join(...)](https://golang.org/src/path/filepath/path.go?s=5893:5925#L199) to keep it generic.

Reply:
Okay. Sure